pubs/commands/note_cmd.py

```diff
@@ -16,6 +18,8 @@ def parser(subparsers, conf):
                         ).completer = CiteKeyCompletion(conf)
     parser.add_argument('-a', '--append',
                         help='append a line of text to the notes file', default=None)
+    parser.add_argument('-e', '--echo', action='store_true',
+                        help='echo the contents of the notes file')
```

Comment:
I can look into that. 
```
$ pubs note citekey
```
Already opens the note file into an editor. Would `--open-with` be duplicating this functionality? 

I see in the `.pubsrc` file the parameter, `edit_cmd` sets which editor to use when editing bibtex files. Does this also work with notes files? If not, perhaps this would keep the command arguments consistent with `pubs note` having similar behavior to `pubs edit` except the open different text files, the pubs/notes/citekey.txt and the pubs/bib citekey.bib.

Reply:
In the rest of pubs, we have one main behavior for bibtex and meta files which is to edit them; and one main behavior for other documents, that is to _open_ them, where you can optionally chose an other program. For example, I use an alias to open with `lp` in order to directly print an article.

I think editing is a special enough that we keep it separate and offer an option to overload the default editor in the config. That would answer your question because changing the `edit_cmd` config would apply to all edition and would always apply, while here I guess sometime you want to edit the note, sometime just to show it.

As for the _show_ or _open_ use case, we keep the default offered by `xdg-open` so it is dependent on the file type. That works well for pdf, djvu, etc.

The note plugin is a bit new here because it has both use cases although so far we only considered the _edit_ one. I think your issue and PR points out clearly that there is a need for another use case for _viewing_, which I think can be covered as we cover _openning_ for the documents. For instance, the my proposal above allows any user to specify how they want to show the note. One example would be, for someone writing notes in markdown, to show it in the man-page format (by calling a pandoc conversion). The nice thing with abstracting it away as a "open with" is that this is all to the convenience of the user. Also, we tried so far, to focus on foundations that would support most users as well as user-defined customization; the "open" implementation would in my opinion be more in line with that.

This was to give the general motivation and philosophy I would see here. That said, I only suggested the `--open-with` approach above because it the simplest and is backward compatible. A more consistent approach here would actually be to introduce sub-commands for `note`  similarly to the rest of pubs (e.g. `pubs note edit xxx`, `pubs note open xxx`) but it would be more work and not backward compatible (which I think would be fine here anyway). From my point of view, it is up to you to chose which one you are OK to implement here.